Use a unique root in logic and fix playthrough inconsistencies

Roman971 opened the pull request:
Mostly internal logic changes, shouldn't change anything other than making generation slightly faster and fixing some playthrough issues in the spoiler.

- Change Link's House and Temple of Time save&quit points to be children of a common Root region.
- Update the playthrough algorithm to support a unique common root, and remove assumptions about Beyond Door of Time always having a path to Link's House.
- Change the Master Sword Pedestal event item to "Time Travel" to make the playthrough consistent between child and adult start.

I had to make some changes to the new playthrough algorithm so I'm curious to hear what @Zannick thinks about it and if he knows a better way to handle having a common root. I think this is good enough as it is but there might be some possible improvements.

Zannick commented on Playthrough.py:
```diff
@@ -116,13 +113,17 @@ def iter_reachable_locations(self, item_locations):
                 adult_failed.extend(Playthrough._expand_regions(
                         adult_queue, adult_regions, validate_adult,
                         child_failed, child_regions))
+            if child_queue:
```
This part isn't actually required: when starting adult, the first _expand_regions call will add Root to the child_queue before the second call; the third call is specific for starting child and gaining Time Travel.

Roman971 replied:
Alright, I fixed it. I added it just in case but I see now it's not needed due to the order of _expand_region calls.

Zannick commented:
I definitely like the approach of a single Root region, and the playthrough algorithm doesn't really care one way or the other. I think this is a very clean way to take care of that assumption.

But this might also let us generalize a little further. I think with this we could make the Playthrough algorithm start with Root accessible in both ages, since it won't be able to progress out of that region in the non-starting age until it gets Time Travel--doing this should let us drop the cross-age stuff in _expand_regions as well as the third extra-special call.

Roman971 commented:
Fixed the unnecessary call as discussed above and updated Glitched Logic to use the unique root. I had forgotten about it when I first opened the PR, still not used to having 2 different sets of logic json files.

Roman971 commented:
@Zannick I removed some of the unnecessary playthrough algorithm code as you suggested. Both ages start with the root now since the non-starting age won't be able to leave it anyway until "Time Travel" is obtained, and the cross age stuff in _expand_regions is also gone.

Zannick commented:
Looks great from my perspective. Thanks!
